### DISCUSSION TITLE
Option to not update localstorage with persist middleware

### DISCUSSION THREAD

**Discussion by aryanshridhar** [pmndrs/zustand#1883] [Q&A] [answered]
Hey folks, not sure if this is plausible as of now or something that y'all have planned in the roadmap; Is it possible to not update the localstorage upon a change in store when using `persist` middleware depending upon a certain value?

To simply this - 
Let's say I have a store which looks like - 
```js
const initialState = {
  shouldUpdateStorage: true,
  data: [],
};
```

What `persist` middleware will do is to update the localstorage when a change in store is occurred. I want this behaviour only when the `shouldUpdateStorage` value in store is set to true. Let's say somewhere, I toggle the store `shouldUpdateStorage` variable to `false`, then is it possible for zustand to not update the localstorage if any change is done in store (or, detach itself from localstorage).

The `shouldUpdateStorage` need not be a store variable, it can be any global variable as well.

**Comment by dbritto-dev** [comment, marked as answer]
@aryanshridhar you can build your own storage and add some validations on `setItem()`
